easypy/threadtree.py

```diff
@@ -258,3 +261,116 @@ def dump_threads():
     thread = concurrent(dump_threads, threadname="ThreadWatch", loop=True, sleep=interval)
     thread.start()
     _logger.info("threads watcher started")
+
+
+class ThreadContexts():
+    """
+    A structure for storing arbitrary data per thread.
+    Unlike threading.local, data stored here will be inherited into
+    child-threads (threads spawned from threads).
+
+        TC = ThreadContexts()
+
+        # (thread-a)
+        with TC(my_data='b'):
+            spawn_thread_b()
+
+        # (thread-b)
+        assert TC.my_data == 'b'
+        with TC(my_data='c'):
+            spawn_thread_c()
+        assert TC.my_data == 'b'
+
+        # (thread-c)
+        assert TC.my_data == 'c'
+
+    'accumulated': attributes named here get incremented each time, instead of overwritten:
+
+        TC = ThreadContexts(accumulated=('i', 'j'))
+        assert TC.i == TC.j == 0
+
+        with TC(i=1):
+            assert TC.i == 1
+            assert TC.j == 0
+
+            with TC(i=1, j=1):
+                assert TC.i == 2
+                assert TC.j == 1
+
+
+    'concatenated': attributes named here get concatenated into a list, instead of overwritten:
```

Comment:
concatenated usually used for strings...  I'd say "merged", "joined", "appended"